src/app/core/session/login/login.component.ts

```diff
@@ -45,64 +44,61 @@ export class LoginComponent {
 
   constructor(
     private _sessionService: SessionService,
-    @Optional() private router: Router,
-    @Optional() private route: ActivatedRoute
+    private loggingService: LoggingService,
+    private router: Router,
+    private route: ActivatedRoute
   ) {}
 
   /**
    * Do a login with the SessionService.
    */
   login() {
     this.loginInProgress = true;
+    this.errorMessage = "";
 
     this._sessionService
       .login(this.username, this.password)
       .then((loginState) => {
-        if (loginState === LoginState.LOGGED_IN) {
-          this.onLoginSuccess();
-        } else {
-          if (
-            this._sessionService.getSyncState().getState() ===
-              SyncState.ABORTED &&
-            this._sessionService.getConnectionState().getState() ===
-              ConnectionState.OFFLINE
-          ) {
+        switch (loginState) {
+          case LoginState.LOGGED_IN:
+            this.onLoginSuccess();
+            break;
+          case LoginState.UNAVAILABLE:
```

Comment:
with the LocalSession making a difference between username wrong (--> UNAVAILABLE) and password wrong (--> FAILED) we shouldn't handle the two states differently for the user in the UI - otherwise this potentially exposes what usernames exist to unauthenticated attackers?

Reply:
I added this for a better feedback when offline. This effect is only visible when offline and it should help to show the user whether just the password was misspelled or if the user is generally not available offline. Which user is available offline can be easily found out by an attacker by looking at the local storage.